crates/bevy_transform/src/systems.rs

```diff
@@ -46,21 +46,28 @@ pub fn propagate_transforms(
             changed |= children_changed;
 
             for child in children.iter() {
-                propagate_recursive(
-                    &global_transform,
-                    &transform_query,
-                    &parent_query,
-                    &children_query,
-                    entity,
-                    *child,
-                    changed,
-                );
+                // SAFETY: Assuming the hierarchy is consistent, we can be sure that each `child` entity
```

Comment:
I don't think this is 100% accurate. `propagate_recursive` ensures consistency internally via the assertion it makes.

What we need to assert is that we don't alias *the entire subtree* of the hierarchy across threads, which is enforced by the root level parallelization.

Reply:
Does it seem correct now?

Reply:
This is better but I think the wording isn't very clear. See the other comment for a better suggestion on how to word it for clarity.